admin/python/lsst/qserv/admin/workerMgmt.py

```diff
@@ -0,0 +1,175 @@
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+
+"""
+Module defining WorkerMgmt class and related methods.
+
+WorkerMgmt class responsibility is to support operations on a set of
+worker nodes, for example creating/deleting databases or tables.
+
+@version $Id$
+
+@author Andy Salnikov - salnikov@slac.stanford.edu
+"""
+
+#--------------------------------
+#  Imports of standard modules --
+#--------------------------------
+import types
+
+#-----------------------------
+# Imports for other modules --
+#-----------------------------
+from .workerAdmin import WorkerAdmin
+
+#----------------------------------
+# Local non-exported definitions --
+#----------------------------------
+
+#------------------------
+# Exported definitions --
+#------------------------
+
+#---------------------
+#  Class definition --
+#---------------------
+
+class WorkerMgmt(object):
+    """
+    WorkerMgmt class main purpose is to facilitate management operations on
+    a set of workers. It's two main responsibities are:
+      1. selecting a corresponding set of worker nodes
+      2. doing some operation on the selected set of nodes
+
+    For some operations latter responsibility can possibly be delegated to some
+    other entity, e.g. WorkerAdmin class.
+    """
+
+    #----------------
+    #  Constructor --
+    #----------------
+    def __init__(self, qservAdmin):
+        """
+        Constructor needs an instance of QservAdmin type which provides access to
+        CSS worker node information.
+
+        @param qservAdmin:  QservAdmin instance.
+        """
+
+        self.css = qservAdmin
+
+    #-------------------
+    #  Public methods --
+    #-------------------
+
+    def select(self, state=None, nodeType=None):
+        """
+        Returns set of WorkerAdmin instances based on supplied selection criteria.
+
+        @param state:   string or list of strings from a list defined in  QservAdmin.NodeState.STATES
+                        (e.g. QservAdmin.NodeState.ACTIVE). If provided then only nodes with
+                        the state that match any item in the list (or the state if state is a string)
+                        are returned.
+        @param nodeType: string or list of strings, if provided then only nodes that have the specified
+                        node type are returned
+        """
+
+        nodes = self.selectDict(state, nodeType)
+
+        # convert to instances
+        return [WorkerAdmin(name=key, qservAdmin=self.css) for key, _ in nodes.items()]
+
+
+    def selectDict(self, state=None, nodeType=None):
```

Comment:
The name is because this method returns dict. All select methods (two of them for now) return "nodes", just in different representation, `select()` returns the set of `WorkerAdmin` instances, `selectDict()` returns dictionary where key is node name and value is another dict. 'selectNodes()` would be too generic name for this method. It is not private.